File: zad1/Calculator.js
module.exports = class Calculator {
    static getGreatestCommonDivisor(a, b) {
        Calculator.throwIfBothZeros(a, b);
        return a == 0 ? b : Calculator.getGreatestCommonDivisor(b % a, a);
    }

    static getLeastCommonMultiple(a, b) {
        Calculator.throwIfNonUniqueInput(a, b);
        return Math.abs(a * b) / Calculator.getGreatestCommonDivisor(a, b);
    }

    static throwIfBothZeros(a, b) {
        if (a == 0 && b == 0) {
            throw "Provided numbers can't be both zeros!"
        }
    }

    static throwIfNonUniqueInput(a, b) {
        if (a == b) {
            throw "Enter two unique numbers";
        }
    }
}

File: zad1/Calculator.py
def getGreatestCommonDivisor(a, b):
    throwIfBothZeros(a, b)
    gcd = b if a % b == 0 else getGreatestCommonDivisor(b, a % b)
    return gcd if int(gcd) == gcd else 1

def getLeastCommonMultiple(a, b):
    throwIfNonUniqueInput(a, b)
    return abs(a * b) / getGreatestCommonDivisor(a, b)


def throwIfBothZeros(a, b):
    if a == 0 and b == 0:
        raise ValueError("Provided numbers can't be both zeros!")

def throwIfNonUniqueInput(a, b):
    if a == b:
        raise ValueError("Enter two unique numbers")
